svm/src/account_loader.rs

```diff
@@ -95,6 +96,264 @@ pub struct FeesOnlyTransaction {
     pub fee_details: FeeDetails,
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct AccountLoader<'a, CB: TransactionProcessingCallback> {
+    pub program_cache: ProgramCacheForTxBatch,
+    account_overrides: Option<&'a AccountOverrides>,
+    account_cache: HashMap<Pubkey, AccountCacheItem>,
+    callbacks: &'a CB,
+}
+impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
+    pub fn new(
+        callbacks: &'a CB,
+        account_overrides: Option<&'a AccountOverrides>,
+        program_cache: ProgramCacheForTxBatch,
+    ) -> AccountLoader<'a, CB> {
+        Self {
+            program_cache,
+            account_overrides,
+            account_cache: HashMap::new(),
```

Comment:
let's start with some reasonable capacity so we re-allocate as little as possible.
maybe 256. it's not going to be perfect but can help us avoid allocating several times per batch.

Reply:
> i wasnt sure the reallocs are actually that expensive

Re-allocs are often the worst for performance, because not only do we need to do an allocation and a deallocation (2 possible yields to os scheduler) we then also need to copy all the data from one to another.  

This is per-batch, so it's not as bad as doing an allocation per transaction - but we should still avoid it the best we can.

Reply:
[824b350](https://github.com/anza-xyz/agave/pull/3146/commits/824b3502de2f0c0b43e101f279cafce73d739e74)